spec/lib/mpi/messages/find_profile_message_fields_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'mpi/messages/find_profile_message_fields'
+
+describe MPI::Messages::FindProfileMessageFields do
+  describe '.valid?' do
+    subject { described_class.new(profile) }
+
+    let(:missing_keys) { %i[given_names last_name birth_date ssn] }
+
+    before do
+      subject.validate
+    end
+
+    context 'missing keys and values' do
+      let(:profile) { {} }
+
+      its(:valid?) { is_expected.to be(false) }
+      its(:missing_keys) { is_expected.to be(true) }
+      its(:missing_values) { is_expected.to be(true) }
+    end
+
+    context 'missing values' do
+      let(:profile) { { given_names: nil, last_name: '', birth_date: nil, ssn: '' } }
+
+      its(:valid?) { is_expected.to be(false) }
+      its(:missing_keys) { is_expected.to be(false) }
+      its(:missing_values) { is_expected.to be(true) }
+    end
+
+    context 'valid-ish' do
```

Comment:
```suggestion
    context 'with valid-ish attributes' do
```

Not sure about what's `valid-ish` here